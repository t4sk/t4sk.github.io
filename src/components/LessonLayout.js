import React from "react"
import PropTypes from "prop-types"
import { Icon, Button, Container } from "semantic-ui-react"
import SEO from "./SEO"
import PageLayout from "./PageLayout"
import Html from "./Html"
import YouTube from "./YouTube"
import SideNav from "./SideNav"
import styles from "./LessonLayout.module.css"

export function LessonLayout(props) {
  const { title, baseUrl, chapters, youTube, html } = props

  // TODO back, next buttons

  return (
    <PageLayout>
      <SEO title={`${title} | Smart Contract Programmer`} />
      <div className={styles.component}>
        <div className={styles.row}>
          <SideNav baseUrl={baseUrl} chapters={chapters} />
          <div className={styles.main}>
            <Container>
              <div className={styles.youTube}>
                {youTube && <YouTube {...youTube} />}
              </div>

              <Html html={html} />

              <div className={styles.buttons}>
                <Button animated size="huge">
                  <Button.Content visible>Back</Button.Content>
                  <Button.Content hidden>
                    <Icon name="arrow left" />
                  </Button.Content>
                </Button>
                <Button animated size="huge">
                  <Button.Content visible>Next</Button.Content>
                  <Button.Content hidden>
                    <Icon name="arrow right" />
                  </Button.Content>
                </Button>
              </div>
            </Container>
          </div>
        </div>
      </div>
    </PageLayout>
  )
}

LessonLayout.propTypes = {
  title: PropTypes.string.isRequired,
  baseUrl: PropTypes.string.isRequired,
  chapters: PropTypes.arrayOf(
    PropTypes.shape({
      header: PropTypes.string.isRequired,
      lessons: PropTypes.array.isRequired,
    })
  ).isRequired,
  youTube: PropTypes.object,
  html: PropTypes.string.isRequired,
}

export default LessonLayout
