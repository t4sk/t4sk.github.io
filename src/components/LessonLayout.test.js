import React from "react"
import { shallow } from "enzyme"
import { LessonLayout } from "./LessonLayout"

const props = {
  title: "Multi-Sig Wallet",
  baseUrl: "http://localhost:3000",
  chapters: [
    {
      header: "Solidity Contract",
      lessons: [
        {
          icon: "youtube",
          color: "red",
          href: "",
          path: "00-overview",
          header: "Intro",
          description: "",
        },
        {
          icon: "github",
          color: "",
          href: "https://github.com/t4sk/solidity-multi-sig-wallet",
          openNewTab: true,
          path: "",
          header: "Code",
        },
      ],
    },
  ],
  youTube: {
    url: "https://youtube.com",
    start: 1,
    end: 123,
  },
  html: "<p>foo bar</p>",
}

beforeEach(() => {})

test("it renders", () => {
  const component = shallow(<LessonLayout {...props} />)

  expect(component).toMatchSnapshot()
})
