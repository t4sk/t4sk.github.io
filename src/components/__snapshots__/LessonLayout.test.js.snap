// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<PageLayout>
  <SEO
    description="Learn how to program smart contracts."
    keywords={
      Array [
        "smart contract",
        "solidity",
        "ethereum",
      ]
    }
    lang="en"
    meta={Array []}
    title="Multi-Sig Wallet | Smart Contract Programmer"
  />
  <div
    className="component"
  >
    <div
      className="row"
    >
      <SideNav
        baseUrl="http://localhost:3000"
        chapters={
          Array [
            Object {
              "header": "Solidity Contract",
              "lessons": Array [
                Object {
                  "color": "red",
                  "description": "",
                  "header": "Intro",
                  "href": "",
                  "icon": "youtube",
                  "path": "00-overview",
                },
                Object {
                  "color": "",
                  "header": "Code",
                  "href": "https://github.com/t4sk/solidity-multi-sig-wallet",
                  "icon": "github",
                  "openNewTab": true,
                  "path": "",
                },
              ],
            },
          ]
        }
      />
      <div
        className="main"
      >
        <Container>
          <div
            className="youTube"
          >
            <YouTube
              end={123}
              start={1}
              url="https://youtube.com"
            />
          </div>
          <Html
            html="<p>foo bar</p>"
          />
          <div
            className="buttons"
          >
            <Button
              animated={true}
              as="button"
              size="huge"
            >
              <ButtonContent
                visible={true}
              >
                Back
              </ButtonContent>
              <ButtonContent
                hidden={true}
              >
                <Icon
                  as="i"
                  name="arrow left"
                />
              </ButtonContent>
            </Button>
            <Button
              animated={true}
              as="button"
              size="huge"
            >
              <ButtonContent
                visible={true}
              >
                Next
              </ButtonContent>
              <ButtonContent
                hidden={true}
              >
                <Icon
                  as="i"
                  name="arrow right"
                />
              </ButtonContent>
            </Button>
          </div>
        </Container>
      </div>
    </div>
  </div>
</PageLayout>
`;
